tla/etcdraft.tla

```diff
@@ -0,0 +1,913 @@
+--------------------------------- MODULE etcdraft ---------------------------------
+\* Copyright 2024 The etcd Authors
+\*
+\* Licensed under the Apache License, Version 2.0 (the "License");
+\* you may not use this file except in compliance with the License.
+\* You may obtain a copy of the License at
+\*
+\*     http://www.apache.org/licenses/LICENSE-2.0
+\*
+\* Unless required by applicable law or agreed to in writing, software
+\* distributed under the License is distributed on an "AS IS" BASIS,
+\* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+\* See the License for the specific language governing permissions and
+\* limitations under the License.
+\*
+\*
+\* This is the formal specification for the Raft consensus algorithm.
+\*
+\* Copyright 2014 Diego Ongaro, 2015 Brandon Amos and Huanchen Zhang,
+\* 2016 Daniel Ricketts, 2021 George Pîrlea and Darius Foo.
+\*
+\* This work is licensed under the Creative Commons Attribution-4.0
+\* International License https://creativecommons.org/licenses/by/4.0/
+
+EXTENDS Naturals, Integers, Bags, FiniteSets, Sequences, SequencesExt, FiniteSetsExt, BagsExt, TLC
+
+\* The initial and global set of server IDs.
+CONSTANTS InitServer, Server
+
+\* Log metadata to distinguish values from configuration changes.
+CONSTANT ValueEntry, ConfigEntry
+
+\* Server states.
+CONSTANTS 
+    \* @type: Str;
+    Follower,
+    \* @type: Str;
+    Candidate,
```

Comment:
I don't see PreCandidate and PreVote included in the spec. Should those be included?

Reply:
For now this is not included in the spec. Per my understanding, it is an optional optimization that do not impact implementation correctness. But we can discuss this further.